lib/react.dart

```diff
@@ -1461,6 +1491,16 @@ class SyntheticWheelEvent extends SyntheticEvent {
   throw new Exception('setClientConfiguration must be called before registerComponent.');
 };
 
+/// Registers [componentFactory] on both client and server.
+/*ComponentRegistrar*/ Function registerComponent2 = (/*ComponentFactory*/ componentFactory,
+    [/*Iterable<String>*/ skipMethods = const [
```

Comment:
Is there a reason that the "version 2" of this can't have all these types? 

I know we had them commented out in "version 1"... but... I feel like we might be able to have typing on this stuff now.